ansible/roles/gpu/tasks/nvidia-gpu.yaml

```diff
@@ -56,12 +56,18 @@
     nvidia_driver_versions: "{{ nvidia_driver_files.files | map(attribute='path') | map('regex_replace', '^.*NVIDIA-Linux-x86_64-(\\d+\\.\\d+\\.\\d+)\\.run$', '\\1') | list }}"
     uninstall_list: []
 
-- name: compare versions with nvidia_driver_version
+- name: compare versions with nvidia_driver_version for uninstall
   loop: "{{ nvidia_driver_versions }}"
-  when: item is version( nvidia_driver_version , '<')
+  when: item is version( nvidia_driver_version , '<=')
   set_fact:
     uninstall_list: "{{ uninstall_list + [item] }}"
 
+- name: compare versions with nvidia_driver_version for removal
+  loop: "{{ nvidia_driver_versions }}"
+  when: item is version( nvidia_driver_version , '<')
+  set_fact:
+    remove_list: "{{ remove_list + [item] }}"
```

Comment:
Do we need a set_fact `remove_list: []` on line 58, like we have with `uninstall_list`?